scripts/apps/workspace/content/directives/ContentCreateDirective.ts

```diff
@@ -56,23 +56,24 @@ export function ContentCreateDirective(
                 return $location.path() !== '/workspace/personal';
             };
 
+            scope.$on('template:update', (e, data) => {
+                if (data?.desks.includes(desks.activeDeskId)) {
+                    getRecentTemplates(desks.activeDeskId);
```

Comment:
When can `data.desks` not be present? Shouldn't templates be updated anyway?

Reply:
>When can data.desks not be present?

Will always be there, will fix it.

Templates could be accessed only from the desk they are allowed to. So, why call `getRecentTemplates` in a desk if a template of another desk is updated?

Reply:
> Templates could be accessed only from the desk they are allowed to. So, why call `getRecentTemplates` in a desk if a template of another desk is updated?

If it's cheap from a performance point of view to update templates unconditionally, I'd do that. We never know what kind of feature requests will come our way so it makes sense not to constrain ourselves too much when possible. For example, a requirement might come in to also display the templates of routed desks and we would get a bug if we forget to change this update logic too.